db-assessment/import_db_assessment.py

```diff
@@ -0,0 +1,833 @@
+
+    print ('\nPreparing to upload CSV files\n')
+
+    # Creating Hash Table with all expected table schemas to be imported
+    tableSchemas = {}
+    tableSchemas = getBQJobConfig()
+
+    # Getting the name of the target table_name to import the data based on the filename from OS
+    for fileName in fileList:
+        
+        # Default Big Query Job Configurations for Optimus Prime CSV files
+        autoDetect = 'True'
+
+        # Final table name from the CSV file names
+        tableName = getObjNameFromFiles(fileName,'__',1)
+
+        print ('\nThe filename {} is being imported to Big Query.'.format(fileName))
+
+        # Import the given CSV fileName into 
+        importCSVToBQ(gcpProjectName,bqDataset,tableName,fileName,skipLeadingRows,autoDetect,tableSchemas)
+
+    return True
+
+def importCSVToBQ(gcpProjectName,bqDataset,tableName,fileName,skipLeadingRows,autoDetect,tableSchemas):
+# This function will import the CSV file into the Big Query using the proper project.dataset.tablename
+# A Big Query Job is created for it
+
+    # Construct a BigQuery client object.
+    client = bigquery.Client(client_info=set_client_info.get_http_client_info())
+
+    # Adding Project and Dataset based on arguments 
+    # table_id to the ID of the table to create.
+    if gcpProjectName is not None:
+        table_id = str(gcpProjectName) + '.' + str(bqDataset) + '.' + str(tableName)
+    
+    # In case projectname was passed as argument. Then, it tries to get the default project for the [service] account being used
+    else:
+        table_id = str(client.project) + '.' + str(bqDataset) + '.' + str(tableName)
+
+    # table schema
+    schema = []
+
+    # Getting table schema
+    try:
+        schema = tableSchemas[tableName]
+    except KeyError:
+        # In case there is not expected table schema found in getBQJobConfig function
+        print ('\nWARNING: The filename {} could not be imported to Big Query.'.format(fileName))
+        print ('The table name {} cannot be imported because it does not have table schema in Optimus Prime configuration. So, it will be skipped.\n')
+        return False
+
+    job_config = bigquery.LoadJobConfig(
+        schema=schema,
+        skip_leading_rows=skipLeadingRows,
+        # The source format defaults to CSV, so the line below is optional.
+        source_format=bigquery.SourceFormat.CSV,
+    )
+
+    with open(fileName, "rb") as source_file:
+        load_job = client.load_table_from_file(source_file, table_id, job_config=job_config)
+
+    load_job.result()  # Waits for the job to complete.
+
+    destination_table = client.get_table(table_id)  # Make an API request.
+    print("Loaded {} rows into: {}".format(destination_table.num_rows,destination_table.reference))
+    print ('The filename {} is successfully imported to Big Query.\n'.format(fileName))
+
+    # returns True if processing is successfully
+    return True
+
+
+def getObjNameFromFiles(fileName,splitterChar,pos):
```

Comment:
Accepted, for next release. Logic needs to be tested a bit further, because at some point we will have to decide if we can move forward (ignore) or stop processing (error exit).

Reply:
Included in the new PR for newest release.